zerver/lib/test_classes.py

```diff
@@ -2500,13 +2503,21 @@ def send_and_test_private_message(
 
         return msg
 
-    def build_webhook_url(self, *args: str, **kwargs: str) -> str:
-        url = self.URL_TEMPLATE
+    def build_webhook_url(self, *args: str, url_template: str | None = None, **kwargs: str) -> str:
```

Comment:
`build_webhook_url()` now updates legacy URLs, using recursion.

Since webhooks use it with `self.url = build_webhook_url(...)`, they just update self.url and now they update self.legacy_urls. But, of course, the code does not feel tight enough as the legacy urls are silently updated while the self.url is only returned. But this was the way I could think of to implement this without having to edit every single webhook.
I'm eager to hear better approaches.